src/aphroditeInterfaceFactory.js

```diff
@@ -45,13 +48,44 @@ function separateStyles(stylesArray) {
   };
 }
 
+function stylesDiff(ltrStyles, rtlStyles) {
+  const styles = {}; // only includes the necessary styles and overrides
+  Object.keys(rtlStyles)
+    .filter(key => rtlStyles[key] !== ltrStyles[key])
+    .forEach((key) => {
+      let value = rtlStyles[key];
+      if (value && typeof value === 'object') {
+        value = stylesDiff(ltrStyles[key], value);
+      }
+      if (value !== null && typeof value !== 'undefined') styles[key] = value;
+    });
+
+  // styles that have been deleted from the LTR object need to be overridden
+  Object.entries(ltrStyles)
+    .filter(([key]) => !rtlStyles[key] && rtlStyles[key] !== ltrStyles[key])
+    .forEach(([key, value]) => {
+      if (typeof value !== 'object') {
+        styles[key] = 'auto';
+      }
+    });
+
+  if (Object.keys(styles).length === 0) return null;
```

Comment:
It'd be less clean, but more performant, if we set a boolean flag whenever we assign to the object, and then checked that here.